backend/resell-dapp-api/src/events/event.service.ts

```diff
@@ -3,13 +3,76 @@ import { Injectable } from '@nestjs/common';
 import { InjectModel } from '@nestjs/mongoose';
 import { Event } from './schemas/event.schema';
 import { CreateEventDto } from './dto/create-event.dto';
+import { TicketDto } from './dto/ticket.dto';
 
 @Injectable()
 export class EventService {
+
   constructor(@InjectModel(Event.name) private eventModel: Model<Event>) {}
 
-  async create(createEventDto: CreateEventDto): Promise<Event> {
+  async createEvent(createEventDto: CreateEventDto): Promise<Event> {
     const createdEvent = new this.eventModel(createEventDto);
     return createdEvent.save();
   }
+
+  async createTicket(ticketDto: TicketDto, contractAddress: string): Promise<Event> {
+    const event = await this.eventModel.findOne({ contractAddress: contractAddress });
+    if (event) {
+      // check if ticket already exists
+      const ticket = event.ticketList.find(ticket => ticket.ticket_address === ticketDto.ticket_address);
+      if (ticket) {
+        throw new Error('Ticket already exists');
```

Comment:
What does throwing error in event.service do? I imagine we want to return meaningful stuff to the caller as well, such as 404 if an event doesn't exist, 500 if there's an unexpected error etc

Reply:
Great comment, you're right! I'm a bit rusty, haven't been programming for the last 9 months ^^". Will try to fix it before our call. Also, tested the apis via swagger and they seem to work as intended. Let's review them together this evening tho